### PR TITLE
Example in docstring of `select_nested_models`

### LINKED ISSUE #4788: Expose undocumented new features from v1.1 and v1.0
Several features have been introduced in v1.0 and v1.1 but lack proper documentation. Ideally this should sorted for 1.2.

The main features are
- [ ] `gammapy.estimators.resample_energy_edges` : the question of providing minimum number of counts of significance per bin comes regularly. This could be exposed in several places (e.g. in an how-to and in some tutorials)
- [ ] `select_nested_models` can be demonstrated for different use cases, e.g. source TS estimation and/or spectral feature detection
- [ ] observation clustering should also be presented in its own notebook.
- [ ] we should expose the `DatasetsMaker`
- [ ] we could demonstrate using different spectral normalization models for `FoVBackgroundModel`

Are there others? 

I propose we discuss possible presentations in this issue's discussion

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
**Description**
This PR is to include an example in the docstring of `select_nested_models`. To resolve #4788

For now I think this is enough to allow users to know how the function can be utilised.
At a later stage we can create a tutorial once some further features have been implemented by @QRemy. 

Note: I also adapted one line in `gammapy.utils.parallel.multiprocessing_manager` to fix the docs.

### CODE REVIEW

**Review by QRemy — CHANGES_REQUESTED** [review]
Thanks @Astro-Kirsty, I have just one comment :

File: gammapy/modeling/selection.py [hunk 1]
```diff
@@ -151,6 +151,36 @@ def select_nested_models(
...
+    ::
+
+        from gammapy.modeling.selection import select_nested_models
+        from gammapy.datasets import Datasets, SpectrumDatasetOnOff
+        from gammapy.modeling.models import SkyModel
+
+        # Test if cutoff is significant
+        dataset = SpectrumDatasetOnOff.read("$GAMMAPY_DATA/joint-crab/spectra/hess/pha_obs23523.fits")
+        datasets = Datasets(dataset)
+        model = SkyModel.create(spectral_model="ecpl", spatial_model="point", name='hess')
```
Comment (QRemy) [inline gammapy/modeling/selection.py:166]:
```suggestion
        model = SkyModel.create(spectral_model="ecpl", spatial_model="point", name='hess')
        model.spatial_model.freeze()
        n_free_parameters = len(model.parameters.free_parameters)
```

**Review by QRemy — APPROVED** [review]
Thanks @Astro-Kirsty, no further comment.

File: gammapy/modeling/selection.py [hunk 2]
```diff
@@ -151,6 +151,42 @@ def select_nested_models(
             * "ts" : fit statistic difference with null hypothesis
             * "fit_results" : results for the best fit
             * "fit_results_null" : fit results for the null hypothesis
+
+    Examples
+    --------
+    ::
```
Comment (registerrier) [inline gammapy/modeling/selection.py:157]:
```suggestion
    .. testcode::
```

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty . Are you sure this goes through doctest this way?

**Pull request merged**